sentry_sdk/integrations/django/caching.py

```diff
@@ -0,0 +1,101 @@
+import functools
+from typing import TYPE_CHECKING
+
+from django import VERSION as DJANGO_VERSION
+from django.core.cache import CacheHandler
+
+from sentry_sdk import Hub
+from sentry_sdk.consts import OP, SPANDATA
+from sentry_sdk._compat import text_type
+
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Callable
+
+
+METHODS_TO_INSTRUMENT = [
+    "get",
+    "get_many",
+]
+
+
+def _instrument_call(cache, method_name, original_method, args, kwargs):
+    # type: (CacheHandler, str, Callable[..., Any], Any, Any) -> Any
+    description = "{} {}".format(method_name, " ".join(args))
+
+    with Hub.current.start_span(op=OP.CACHE, description=description) as span:
+        cache._sentry_recording = True
+        value = original_method(*args, **kwargs)
+        cache._sentry_recording = False
+
+        if value:
+            span.set_data(SPANDATA.CACHE_HIT, True)
+
+            size = len(text_type(value).encode("utf-8"))
+            span.set_data(SPANDATA.CACHE_ITEM_SIZE, size)
+
+        else:
+            span.set_data(SPANDATA.CACHE_HIT, False)
+
+        return value
+
+
+def _patch_cache_method(cache, method_name):
+    # type: (CacheHandler, str) -> None
+    original_method = getattr(cache, method_name)
+
+    @functools.wraps(original_method)
+    def sentry_method(*args, **kwargs):
+        # type: (*Any, **Any) -> Any
+
+        # If this call is being made inside another call, don't instrument it.
+        if cache._sentry_recording:
+            return original_method(*args, **kwargs)
+        else:
+            return _instrument_call(cache, method_name, original_method, args, kwargs)
+
+    setattr(cache, method_name, sentry_method)
+
+
+def _patch_cache(cache):
+    # type: (CacheHandler) -> None
+    from sentry_sdk.integrations.django import DjangoIntegration
+
+    integration = Hub.current.get_integration(DjangoIntegration)
+    if integration and integration.cache_spans:
```

Comment:
can you move this to `patch_caching` and early return there?

Reply:
In `patch_caching` the integration is always None (I guess because this is run in `setup_once()`. But I have now moved the check right before the calls to `_patch_cache`. So it is now not patching if not needed.